k2/python/k2/autograd.py

```diff
@@ -186,6 +182,40 @@ def backward(
         return None, None, None, None, None, None, grad_a, grad_b
 
 
+class _IndexSelectFunction(torch.autograd.Function):
+
+    @staticmethod
+    def forward(ctx, src: torch.Tensor, index: torch.Tensor) -> torch.Tensor:
+        '''Returns a new tensor which indexes the input tensor along dimension 0
+        using the entries in `index`.
+
+        If the entry in `index` is -1, then the corresponding entry in the
+        returned tensor is 0.
+
+        Caution:
+          `index.dtype == torch.int32` and `index.ndim == 1`.
+
+        Args:
+          src:
+            The input tensor. Either 1-D or 2-D with dtype torch.int32 or torch.float32.
+          index:
+            1-D tensor of dtype torch.int32 containing the indexes to index. If an entry
+            is -1, the corresponding entry in the returned value is 0.
+
+        Returns:
+          The output tensor.
```

Comment:
Saying it returns "the output tensor" is not informative at all.  You could say
 ```
    A tensor with shape index.shape+src.shape[1:] and dtype the same as src,
    e.g. if src.ndim == 1 ans.shape would be (index.shape[0]), if src.ndim==2 ans.shape
    would be (index.shape[0],src.shape[1]).    Will satisfy ans[i] == src[index[i]] if 
   src.ndim == 1, or ans[i,j] == src[index[i],j] if src.ndim == 2, except for entries 
   where index[i] == -1 which will be zero.
```
   Notice how what I say is unambiguous because i'm using code.

Reply:
Fixed.